tests/unit/test_query_language.py

```diff
@@ -176,6 +176,19 @@ class some_table(EventFrame):
     assert isinstance(some_table.some_str, StrEventSeries)
 
 
+def test_construct_respects_custom_table_name():
+    @construct
```

Comment:
Yes, I agree that the status quo is undesirable. My original thought was that all the logic in this decorator can be moved to `Frame.__init__` and the decorator itself just becomes:
```py
def construct(cls: type[T]) -> T:
    return cls()
```
In which case I think it would deserve its name. The fact that we need the decorator at all is pretty irksome, so I'd like it to be as simple as possible. If for any reason we don't want to go down that route then I'd be happy to rename it.

Reply:
As long as this is a live question in your mind I'm happy to leave it to you to improve as and when you see fit

Reply:
+1 for `@table`